src/main/java/step2/StringAddCalculator.java

```diff
@@ -0,0 +1,57 @@
+package step2;
+
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    private static final int DEFAULT_EMPTY_RETURN = 0;
+    private static final String DEFAULT_DELIMITER_REGEX = ",|:";
+    private static final String CUSTOM_DELIMITER_REGEX  = "//(.)\n(.*)";
+    private static final String POSITIVE_NUMBER_REGEX = "^[0-9]+$";
+    private static final String NOT_POSITIVE_NUMBER = "0~9 사이의 숫자가 아닙니다.";
+    private static final Pattern CUSTOM_DELIMITER_PATTERN = Pattern.compile(CUSTOM_DELIMITER_REGEX);
+
+
+
+    public static int splitAndSum(String text) {
+        if (isStringNullOrEmpty(text)) {
+            return DEFAULT_EMPTY_RETURN;
+        }
+        String[] splitNumbers = splitNumber(text);
+        validateNumbersPositive(splitNumbers);
+        return addAllNumbers(splitNumbers);
+    }
+
+    private static void validateNumbersPositive(String[] splitNumbers) {
+        boolean isValidate = Arrays.stream(splitNumbers).allMatch(number->number.matches(POSITIVE_NUMBER_REGEX));
+        if (!isValidate){
+            throw new RuntimeException(NOT_POSITIVE_NUMBER);
```

Comment:
```suggestion
            throw new IllegalArgumentException(NOT_POSITIVE_NUMBER);
```
RuntimeException은 너무 포괄적인 예외인 것 같아요.

IllegalArgumentException과 같이 구체적인 예외를 던져주는것이 더 나아보입니다!

Reply:
넵 근환님, 말씀하신대로, Exception을 구체화 하는게 더 맞는것 같아서 수정하였습니다. 감사합니다.